### DISCUSSION TITLE
I want to know how to use frida-il2cpp-bridge

### DISCUSSION THREAD

**Discussion by daeinlee** [vfsfitvnm/frida-il2cpp-bridge#150] [Q&A]
Hello.
Thanks for making such a great tool.
First of all, I would appreciate it if you note that the text may be a bit odd since I used a translator to write the question.

I often use Frida for work, and I have mainly used it as a JavaScript file using frida tools.
Global-metadata.dat is looking for a decryption method, and I found the frida-il2cpp-bridge that does not need global-metadata.dat and is looking at how to use it, but I do not know how to use it due to my lack of knowledge.
So, I prepared a few questions, and if there are answers, it will be easier to understand.

1. Does frida-il2cpp-bridge use node frida (npm install frida)?

2. Can I know how to use it? I want to know how to run the code, not what it is. (ex: frida -U -l hook.js -f com.android.chrome --no-pause )

3. When the code runs, is it correct to send a log with metadata information when the package's unity-related code is executed?

Thank you very much for your reply.

**Comment by vfsfitvnm** [comment]
Hi!

> I have mainly used it as a JavaScript file using frida tools.

This is the same; but I suggest you to [use TypeScript](https://www.youtube.com/watch?v=9cr4gOPFN4o), it's very helpful and I only support TS users.

Shortly:
1) Not necessarily, I use `frida` or `frida-inject` directly - however you must generate a single `.js` file using `frida-compile` or whatever bundler you prefer;
2) Please **read carefully** [here](https://github.com/vfsfitvnm/frida-il2cpp-bridge/wiki/Installation) and [here - outdated, but helpful](https://hackcatml.tistory.com/94);
3) I'm not sure what you mean - however the answer is you can pretty much do anything you want.

So, you have to:
1) Understand what is a TS project and how you can create one;
2) Understand that `frida-compile` transforms multiple TS file into a single JS one
3) Call `frida` with the generated file (you already know this step)

PS: I'll move this issue to the "discussions" section

**Reply by RaidenRespawnStaff** [reply]
if i were to want to use this on a windows game?
because all the tutorial are made for unity game on android....

**Comment by daeinlee** [comment]
Thank you very much. I will try again
